app/components/Operator/OperatorForm.tsx

```diff
@@ -1,19 +1,125 @@
+import { useDeleteFormChange } from "mutations/FormChange/deleteFormChange";
+import { useRouter } from "next/router";
+import { useCommitFormChange } from "mutations/FormChange/commitFormChange";
+import SavingIndicator from "components/Form/SavingIndicator";
+import { useUpdateFormChange } from "mutations/FormChange/updateFormChange";
+
+interface Props extends FormComponentProps {
+  formChange: OperatorForm_formChange$key;
+}
+
+const OperatorForm: React.FC<Props> = (props) => {
+  const formChange = useFragment(
+    graphql`
+      fragment OperatorForm_formChange on FormChange {
+        id
+        rowId
+        newFormData
+        formDataRecordId
+      }
+    `,
+    props.formChange
+  );
+
+  const router = useRouter();
+
+  const [commitFormChange, isCommittingFormChange] = useCommitFormChange();
+  const [deleteFormChange, isDeletingFormChange] = useDeleteFormChange();
+  const [updateFormChange, isUpdatingFormChange] = useUpdateFormChange();
+
+  const { newFormData } = formChange;
+  const isEditing = formChange.formDataRecordId !== null;
+
+  const handleChange = ({ formData }) => {
+    updateFormChange({
+      variables: {
+        input: {
+          rowId: formChange.rowId,
+          formChangePatch: {
+            newFormData: formData,
+          },
+        },
+      },
+      optimisticResponse: {
+        updateFormChange: {
+          formChange: {
+            id: formChange.id,
+            newFormData: formData,
+            changeStatus: "pending",
+          },
+        },
+      },
+      debounceKey: formChange.id,
+    });
+  };
+  const handleDiscard = () => {
+    deleteFormChange({
+      variables: {
+        input: {
+          id: formChange.id,
+        },
+      },
+      onCompleted: () => {
+        return router.push(getOperatorsPageRoute());
+      },
+    });
+  };
+
+  const handleSubmit = ({ formData }) => {
+    commitFormChange({
+      variables: {
+        input: {
+          rowId: formChange.rowId,
+          formChangePatch: {
+            newFormData: formData,
+          },
+        },
+      },
+      onCompleted: () => {
+        return router.push(getOperatorsPageRoute());
+      },
+    });
+  };
 
-const OperatorForm: React.FC<FormPageFactoryComponentProps> = (props) => {
   return (
-    <FormBase {...props} schema={operatorSchema as JSONSchema7} uiSchema={{}}>
-      <Button type="submit" style={{ marginRight: "1rem" }}>
-        Submit
-      </Button>
-      <Button type="button" variant="secondary" onClick={props.onDiscard}>
-        Discard Changes
-      </Button>
-    </FormBase>
+    <>
+      {" "}
+      <header>
+        <h2>
+          {isEditing ? "Edit" : "New"} {"Operator"}
+        </h2>
+        <SavingIndicator
+          isSaved={!isUpdatingFormChange && !isCommittingFormChange}
+        />
+      </header>
+      <FormBase
+        {...props}
+        id="operator-form"
+        formData={newFormData}
+        schema={props.schema}
```

Comment:
why don't we pass a "uiSchema" prop?

Reply:
not needed for operator, as none of the forms have any unique ui